app/models/manageiq/providers/amazon/cloud_manager/ssa_agent_manager.rb

```diff
@@ -256,63 +242,87 @@ def get_key_pair(pair_name = MIQ_SSA)
 
   def create_pem_file(pair_name = MIQ_SSA)
     kp = get_key_pair(pair_name)
-    pem_file_name = pair_name+".pem"
+    pem_file_name = pair_name + ".pem"
     File.open(pem_file_name, 'w') {|f| f.write(kp.auth_key) }
     File.chmod(0400, pem_file_name)
     pem_file_name
   end
 
-  def agent_startup_script(script_name = '/tools/amazon_ssa_extract.rb')
-    ssa_root = Gem.loaded_specs['amazon_ssa_support'].full_gem_path
-    ssa_script = ssa_root + script_name
-    ssa_cmd = "ruby #{ssa_script}"
+  def setup_ruby(log = '/var/log/miq_ssa_deploy.log')
+    <<~HEREDOC
```

Comment:
There's nothing actually wrong with naming your Here Doc "HEREDOC" but its kind of like naming a variable "VARIABLE".  You should probably name this and the ones in the other methods something more descriptive to the purpose of the script.